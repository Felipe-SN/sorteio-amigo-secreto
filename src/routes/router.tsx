import Cabecalho from 'components/Cabecalho';
import Configuracao from 'pages/Configuracao';
import Sorteio from 'pages/Sorteio';
import { createBrowserRouter } from 'react-router-dom';

const router = createBrowserRouter([
  {
    element: <Cabecalho />,
    path: '/',
    errorElement: <h2>{'Erro 404!!😵‍💫 Recurso solicitado não encontrado'}</h2>,
    children: [
      { index: true, element: <Configuracao /> },
      { path: 'sorteio', element: <Sorteio /> },
    ],
  },
]);

export default router;
