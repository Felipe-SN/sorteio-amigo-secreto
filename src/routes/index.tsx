import { RouterProvider } from 'react-router-dom';
import router from './router';

export default function Routes() {
  return <RouterProvider router={router} />;
}
