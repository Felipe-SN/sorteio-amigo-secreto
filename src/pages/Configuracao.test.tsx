import 'jest-styled-components';
import { createMemoryRouter, RouterProvider } from 'react-router-dom';
import { RecoilRoot } from 'recoil';
import { render } from '@testing-library/react';
import { routesConfig } from 'routes/router';
import Configuracao from './Configuracao';

const router = createMemoryRouter(routesConfig, {
  initialEntries: ['/'],
});

describe('A pagina de configuração', () => {
  test('Deve ser renderizada corretamente', () => {
    <RouterProvider router={router} />;
    const { container } = render(
      <RecoilRoot>
        <Configuracao />
      </RecoilRoot>
    );

    expect(container).toMatchSnapshot();
  });
});
