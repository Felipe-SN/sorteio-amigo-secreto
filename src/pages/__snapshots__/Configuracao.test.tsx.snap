// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`A pagina de configuração Deve ser renderizada corretamente 1`] = `
<div>
  <div
    class="sc-bcXHqe ldcdWJ"
  >
    <section>
      <h2>
        Vamos começar!
      </h2>
      <form>
        <div
          class="sc-gswNZR heeYVX"
        >
          <input
            placeholder="Insira os nomes dos participantes"
            type="text"
            value=""
          />
          <button
            disabled=""
            type="submit"
          >
            Adicionar
          </button>
        </div>
      </form>
      <ul />
      <footer
        class="sc-eDvSVe flOdzN"
      >
        <button
          class="sc-hLBbgP fXtuGw"
          disabled=""
        >
          Iniciar brincadeira!
        </button>
        <img
          alt="Sacolas de compras"
          aria-hidden="true"
          class="sc-jSUZER eHNkSE"
          src="/img/sacolas.png"
        />
      </footer>
    </section>
  </div>
</div>
`;
